diff --git a/src/loader/loader.c b/src/loader/loader.c
index 0f799c14fd8..b2354ab8f44 100644
--- a/src/loader/loader.c
+++ b/src/loader/loader.c
@@ -556,6 +556,22 @@ loader_open_driver(const char *driver_name,
       snprintf(path, sizeof(path), "%.*s/tls/%s_dri.so", len, p, driver_name);
       driver = dlopen(path, RTLD_NOW | RTLD_GLOBAL);
 #endif
+      if (driver == NULL && __builtin_cpu_supports("avx512bw")) {
+         snprintf(path, sizeof path,
+                  "%.*s/%s_dri.so.avx512", len, p, driver_name);
+         driver = dlopen(path, RTLD_NOW | RTLD_GLOBAL);
+         if (driver == NULL)
+            log_(_LOADER_DEBUG, "MESA-LOADER: failed to open %s: %s\n",
+                 path, dlerror());
+      }
+      if (driver == NULL && __builtin_cpu_supports("avx2") ) {
+         snprintf(path, sizeof path,
+                  "%.*s/%s_dri.so.avx2", len, p, driver_name);
+         driver = dlopen(path, RTLD_NOW | RTLD_GLOBAL);
+         if (driver == NULL)
+            log_(_LOADER_DEBUG, "MESA-LOADER: failed to open %s: %s\n",
+                 path, dlerror());
+      }
       if (driver == NULL) {
          snprintf(path, sizeof(path), "%.*s/%s_dri.so", len, p, driver_name);
          driver = dlopen(path, RTLD_NOW | RTLD_GLOBAL);
