--- 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/nouveau/nvc0/nvc0_resource.h~	2017-09-15 20:06:56.000000000 +0000
+++ 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/nouveau/nvc0/nvc0_resource.h	2017-09-16 16:38:02.398358873 +0000
@@ -34,7 +34,7 @@
 nvc0_miptree_create(struct pipe_screen *pscreen,
                     const struct pipe_resource *tmp);
 
-const struct u_resource_vtbl nvc0_miptree_vtbl;
+extern const struct u_resource_vtbl nvc0_miptree_vtbl;
 
 struct pipe_surface *
 nvc0_miptree_surface_new(struct pipe_context *,
--- 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/auxiliary/util/u_debug_stack.c~	2017-09-15 20:06:56.000000000 +0000
+++ 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/auxiliary/util/u_debug_stack.c	2017-09-16 16:52:38.236779783 +0000
@@ -46,7 +46,7 @@
 #include "os/os_thread.h"
 #include "u_hash_table.h"
 
-struct util_hash_table* symbols_hash;
+static struct util_hash_table* symbols_hash;
 static mtx_t symbols_mutex = _MTX_INITIALIZER_NP;
 
 static unsigned hash_ptr(void* p)
--- 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/radeon/radeon_vce.h~	2017-09-15 20:06:56.000000000 +0000
+++ 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/radeon/radeon_vce.h	2017-09-16 17:07:26.608011430 +0000
@@ -444,7 +444,7 @@
 void radeon_vce_52_init(struct rvce_encoder *enc);
 
 /* version specific function for getting parameters */
-void (*get_pic_param)(struct rvce_encoder *enc,
+extern void (*get_pic_param)(struct rvce_encoder *enc,
                       struct pipe_h264_enc_picture_desc *pic);
 
 /* get parameters for vce 40.2.2 */
--- 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/radeon/radeon_vce.c~	2017-09-15 20:06:56.000000000 +0000
+++ 979978ee06867a531b8d56cee252f5c83920a339/src/gallium/drivers/radeon/radeon_vce.c	2017-09-16 17:25:15.780195234 +0000
@@ -54,6 +54,8 @@
 #define FW_52_8_3 ((52 << 24) | (8 << 16) | (3 << 8))
 #define FW_53 (53 << 24)
 
+void (*get_pic_param)(struct rvce_encoder *enc, struct pipe_h264_enc_picture_desc *pic);
+
 /**
  * flush commands to the hardware
  */
