--- 0ac0e32ce1bff834ad09b1926fbf7f4111b3106a/configure.ac~	2017-09-25 17:01:22.000000000 +0000
+++ 0ac0e32ce1bff834ad09b1926fbf7f4111b3106a/configure.ac	2017-09-27 15:20:24.849455385 +0000
@@ -2564,7 +2564,7 @@
                     ;;
                 xskx)
                     swr_require_cxx_feature_flags "SKX" "defined(__AVX512F__) && defined(__AVX512BW__)" \
-                        ",-target-cpu=x86-skylake,-march=skylake-avx512,-xCORE-AVX512" \
+                        ",-target-cpu=skylake,-march=skylake-avx512" \
                         SWR_SKX_CXXFLAGS
                     AC_SUBST([SWR_SKX_CXXFLAGS])
                     HAVE_SWR_SKX=yes
--- 14cc8c55eadfe66965c81155f8eecdc353df4c14/configure.ac~	2018-04-07 23:16:39.000000000 +0000
+++ 14cc8c55eadfe66965c81155f8eecdc353df4c14/configure.ac	2018-04-07 23:25:07.741991240 +0000
@@ -2560,7 +2560,7 @@
     for opts in $option_list
     do
         unset IFS
-        CXXFLAGS="$opts $save_CXXFLAGS"
+        CXXFLAGS="$save_CXXFLAGS $opts"
         AC_COMPILE_IFELSE(
             [AC_LANG_PROGRAM(
                 [   #if !($preprocessor_test)
--- 14cc8c55eadfe66965c81155f8eecdc353df4c14/src/gallium/drivers/swr/Makefile.am~	2018-04-07 20:45:00.000000000 +0000
+++ 14cc8c55eadfe66965c81155f8eecdc353df4c14/src/gallium/drivers/swr/Makefile.am	2018-04-07 23:46:11.676209728 +0000
@@ -298,7 +298,7 @@
 	$(PTHREAD_CFLAGS) \
 	$(SWR_SKX_CXXFLAGS) \
 	-DKNOB_ARCH=KNOB_ARCH_AVX512 \
-	$(COMMON_CXXFLAGS)
+	$(COMMON_CXXFLAGS) $(SWR_SKX_CXXFLAGS)
 
 libswrSKX_la_SOURCES = \
 	$(COMMON_SOURCES)
--- 14cc8c55eadfe66965c81155f8eecdc353df4c14/src/gallium/drivers/swr/Makefile.am~	2018-04-07 23:46:11.000000000 +0000
+++ 14cc8c55eadfe66965c81155f8eecdc353df4c14/src/gallium/drivers/swr/Makefile.am	2018-04-08 00:03:59.800451647 +0000
@@ -295,10 +295,10 @@
 
 if HAVE_SWR_SKX
 libswrSKX_la_CXXFLAGS = \
-	$(PTHREAD_CFLAGS) \
+	$(COMMON_CXXFLAGS) $(PTHREAD_CFLAGS) \
 	$(SWR_SKX_CXXFLAGS) \
 	-DKNOB_ARCH=KNOB_ARCH_AVX512 \
-	$(COMMON_CXXFLAGS) $(SWR_SKX_CXXFLAGS)
+	$(SWR_SKX_CXXFLAGS) -march=skylake-avx512
 
 libswrSKX_la_SOURCES = \
 	$(COMMON_SOURCES)
