--- 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/intel_extensions.c~	2017-01-07 00:44:29.000000000 +0000
+++ 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/intel_extensions.c	2017-01-08 04:27:06.301211224 +0000
@@ -286,7 +286,7 @@
    if (ctx->API != API_OPENGL_CORE)
       ctx->Extensions.ARB_color_buffer_float = true;
 
-   if (ctx->Mesa_DXTn || driQueryOptionb(&brw->optionCache, "force_s3tc_enable"))
+   if (ctx->Mesa_DXTn || driQueryOptionb(&brw->optionCache, "force_s3tc_enable") || 1)
       ctx->Extensions.EXT_texture_compression_s3tc = true;
 
    ctx->Extensions.ANGLE_texture_compression_dxt = true;
