--- 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_program_cache.c~	2017-01-07 00:44:29.000000000 +0000
+++ 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_program_cache.c	2017-01-07 22:01:05.231031351 +0000
@@ -342,7 +342,7 @@
 
    cache->brw = brw;
 
-   cache->size = 7;
+   cache->size = 37;
    cache->n_items = 0;
    cache->items =
       calloc(cache->size, sizeof(struct brw_cache_item *));
--- 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_vs_surface_state.c~	2017-01-07 00:44:29.000000000 +0000
+++ 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_vs_surface_state.c	2017-01-07 22:57:48.329693180 +0000
@@ -49,7 +49,7 @@
  * Compare this path to brw_curbe.c for gen4/5 push constants, and
  * gen6_vs_state.c for gen6+ push constants.
  */
-void
+__attribute__((optimize("unroll-loops"))) __attribute__ ((hot)) void
 brw_upload_pull_constants(struct brw_context *brw,
                           GLbitfield64 brw_new_constbuf,
                           const struct gl_program *prog,
--- 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_vs_surface_state.c~	2017-01-07 22:57:48.000000000 +0000
+++ 45912fb908f7a1d2efbce0f1dbe81e5bc975fbe1/src/mesa/drivers/dri/i965/brw_vs_surface_state.c	2017-01-07 23:18:17.375839746 +0000
@@ -81,7 +81,13 @@
 
    STATIC_ASSERT(sizeof(gl_constant_value) == sizeof(float));
 
-   for (i = 0; i < prog_data->nr_pull_params; i++) {
+   for (i = 0; i < (prog_data->nr_pull_params & ~3); i+=4 ) {
+      constants[i] = *prog_data->pull_param[i];
+      constants[i + 1] = *prog_data->pull_param[i + 1];
+      constants[i + 2] = *prog_data->pull_param[i + 2];
+      constants[i + 3] = *prog_data->pull_param[i + 3];
+   }
+   for (; i < prog_data->nr_pull_params; i++) {
       constants[i] = *prog_data->pull_param[i];
    }
 
